tests/importer/importer.cpp

```diff
@@ -0,0 +1,761 @@
+
+    // Add models which have imports to the Importer by resolving them. They are automatically saved into the
+    // Importer's library, along with any dependencies, under the names used to resolve them.
+    importer->resolveImports(importedUnits, resourcePath("importer/"));
+    importer->resolveImports(importedComponents, resourcePath("importer/"));
+
+    EXPECT_FALSE(importedUnits->hasUnresolvedImports());
+    EXPECT_FALSE(importedComponents->hasUnresolvedImports());
+
+    // Now create a funky model that imports from all over the place and see what happens.
+    std::string funkyString = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                              "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"funky\">\n"
+                              // Status quo functionality, key is local URL + base file path as the local URL doesn't exist as a key.
+                              "  <import xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"units_imported.cellml\">\n"
+                              "    <units units_ref=\"units1_imported\" name=\"units1FromLibrary\"/>\n"
+                              "  </import>\n"
+                              // Use the local URL for a hidden child source component.
+                              "  <import xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"components_source.cellml\">\n"
+                              "    <component component_ref=\"component1\" name=\"component1FromLibrary\"/>\n"
+                              "  </import>\n"
+                              // Use the absolute URL for a hidden child source component.
+                              "  <import xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\""
+                              + resourcePath("importer/components_source.cellml")
+                              + "\">\n"
+                                "    <component component_ref=\"component2\" name=\"component2FromLibrary\"/>\n"
+                                "  </import>\n"
+                                // Remote website URL used for key.
+                                "  <import xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"https://www.example.com/myComponents.cellml\">\n"
+                                "    <component component_ref=\"component3\" name=\"component3FromLibrary\"/>\n"
+                                "  </import>\n"
+                                // Currently non-existent file used for the key. For this CellML file to function outside of the
+                                // the Importer instance, this model needs to be written to this location.
+                                "  <import xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"i_dont_exist_yet.cellml\">\n"
+                                "    <units units_ref=\"units4\" name=\"units4FromLibrary\"/>\n"
+                                "  </import>\n"
```

Comment:
Same here: how do we expect the importer to resolve this?

Reply:
I take this one back, I can see that there is an entry for `https://www.example.com/myComponents.cellml` in the importer.

Reply:
The `Importer` doesn't have to resolve it until asked.  That's the point.  At that time, the file will either exist in that location (in which case everything is fine), or it will return an error.  This way, the `Importer` can be setup such that switching parameter files in and out of a folder is all that's needed to run different experiments.

Reply:
If anything, I would probably expect a warning or something since we can resolve that import...?

Reply:
> The `Importer` doesn't have to resolve it until asked. That's the point. At that time, the file will either exist in that location (in which case everything is fine), or it will return an error. This way, the `Importer` can be setup such that switching parameter files in and out of a folder is all that's needed to run different experiments.

The file doesn't exist here, so why doesn't the importer generate an issue? Or am I missing something?